gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1280 @@
+            are supposed to train 0 models.
+
+            to get maximum performance, set to the number of your cores, if non-parallelized models are being used in
+            the ensemble (LdaModel).
+
+            For LdaMulticore, the performance gain is small and gets larger for a significantly smaller corpus.
+            In that case, ensemble_workers=2 can be used.
+
+        """
+        # the way random_states is handled needs to prevent getting different results when multiprocessing is on,
+        # or getting the same results in every lda children. so it is solved by generating a list of state seeds before
+        # multiprocessing is started.
+        random_states = [self.random_state.randint(self._MAX_RANDOM_STATE) for _ in range(num_models)]
+
+        # each worker has to work on at least one model.
+        # Don't spawn idle workers:
+        workers = min(ensemble_workers, num_models)
+
+        # create worker processes:
+        # from what I know this is basically forking with a jump to a target function in each child
+        # so modifying the ensemble object will not modify the one in the parent because of no shared memory
+        processes = []
+        pipes = []
+        num_models_unhandled = num_models  # how many more models need to be trained by workers?
+
+        for i in range(workers):
+            try:
+                parentConn, childConn = Pipe()
+                num_subprocess_models = 0
+                if i == workers - 1:  # i is a index, hence -1
+                    # is this the last worker that needs to be created?
+                    # then task that worker with all the remaining models
+                    num_subprocess_models = num_models_unhandled
+                else:
+                    num_subprocess_models = int(num_models_unhandled / (workers - i))
+
+                # get the chunk from the random states that is meant to be for those models
+                random_states_for_worker = random_states[-num_models_unhandled:][:num_subprocess_models]
+
+                p = Process(target=self._generate_topic_models,
+                            args=(num_subprocess_models, random_states_for_worker, childConn))
+
+                processes += [p]
+                pipes += [(parentConn, childConn)]
+                p.start()
+
+                num_models_unhandled -= num_subprocess_models
+
+            except ProcessError:
+                logger.error("could not start process {}".format(i))
+                # close all pipes
+                for p in pipes:
+                    p[1].close()
+                    p[0].close()
+                # end all processes
+                for p in processes:
+                    if p.is_alive():
+                        p.terminate()
+                    del p
+                # stop
+                raise
+
+        # aggregate results
+        # will also block until workers are finished
+        for p in pipes:
```

Comment:
Again, more helpful variable names, and you don't have to add comments to explain which of p[0] and p[1] is the parent and child.

```suggestion
        for (parent_conn, child_conn) in pipes:
```

Reply:
Done